packages/terra-consumer-nav/src/components/user-profile/UserProfile.jsx

```diff
@@ -0,0 +1,192 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import Arrange from 'terra-arrange';
+import Grid from 'terra-grid';
+import Popup from 'terra-popup';
+import Overlay from 'terra-overlay';
+import ResponsiveElement from 'terra-responsive-element';
+import IconClose from 'terra-icon/lib/icon/IconClose';
+import IconEllipses from 'terra-icon/lib/icon/IconEllipses';
+import ProfileLinks from './ProfileLinks';
+import HelpModal from './HelpModal';
+import styles from './UserProfile.scss';
+
+const cx = classNames.bind(styles);
+
+const translations = {
+  signout: 'Sign Out',
+  settings: 'Settings',
+  help: 'Help',
+};
+
+const propTypes = {
+  name: PropTypes.string.isRequired,
+  avatar: PropTypes.oneOfType([
+    PropTypes.string,
+    PropTypes.element,
+  ]),
+  signoutUrl: PropTypes.string.isRequired,
+  profileLinks: PropTypes.arrayOf(PropTypes.shape({
+    slug: PropTypes.string,
+    nav_type: PropTypes.oneOf(['GROUPING', 'EXTERNAL', 'APPLICATION', 'MODAL']),
+    text: PropTypes.string.isRequired,
+    uri: PropTypes.string.isRequired,
+    icon: PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.element,
+    ]),
+  })),
+  help: PropTypes.arrayOf(PropTypes.shape({
+    slug: PropTypes.string,
+    nav_type: PropTypes.oneOf(['GROUPING', 'EXTERNAL', 'APPLICATION', 'MODAL']),
+    target: PropTypes.string,
+    text: PropTypes.string.isRequired,
+    uri: PropTypes.string.isRequired,
+    icon: PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.element,
+    ]),
+    children: PropTypes.arrayOf(PropTypes.shape({
+      slug: PropTypes.string,
```

Comment:
you are using this hash shape a lot. split it off into one constant and set the shape to it:
`children: PropTypes.arrayOf(PropTypes.shape(profileHashShape)),`